partner_communication_switzerland/models/partner_communication.py

```diff
@@ -342,8 +344,17 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
                 'group_id.contract_ids').filtered(
                 lambda s: s.state == 'active')
 
+        make_payment_pdf = True
+        for sponsorship in sponsorships:
```

Comment:
The logic of your code is working and seems good. However you can take advantage of the recordsets methods in order to avoid the loop and all these lines of code. I encourage you to read the [Odoo doc about recordsets](https://www.odoo.com/documentation/10.0/reference/orm.html#recordsets) (especially the mapped function which is very powerful) in order to know them well.